core/commonMain/src/kotlinx/serialization/SerializationExceptions.kt

```diff
@@ -133,3 +133,10 @@ internal constructor(message: String?) : SerializationException(message) {
     // This constructor is used by the generated serializers
     constructor(index: Int) : this("An unknown field for index $index")
 }
+
+/**
+ * Thrown when a map deserializer encounters a repeated map key (and configuration disallows this.)
+ */
+@ExperimentalSerializationApi
+public class DuplicateMapKeyException(key: Any?) :
+    SerializationException("Duplicate keys not allowed in maps. Key appeared twice: $key")
```

Comment:
TODO: Just take a `message` param; I'm probably not going to end up doing assertions on key, and this is going to be annoying to generalize to sets, classes, etc.